### PR TITLE
Fix double-checked locking

### PR DESCRIPTION
**Pull request by Nowheresly**
This pull request fixes a broken double-check locking issue in class LazyMember.
A detailed explanation can be found here:
http://stackoverflow.com/questions/3578604/how-to-solve-the-double-checked-locking-is-broken-declaration-in-java

Contains also some minor performance optimizations

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Nowheresly__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by serious6** [comment]
@Nowheresly this ought to be a good candidate for adding some unit-tests

**Comment by vbauer** [comment]
:+1:

**Comment by Nowheresly** [comment]
Makes sense indeed ;-)
New PR in progress...

**Comment by Nowheresly** [comment]
Unit test added for LazyMember class.
Review or comment greatly appreciated!

**Comment by Nowheresly** [comment]
The second PR is related to the issue 157.

**Comment by serious6** [comment]
@Nowheresly thanks for the contribution. Please split this into 2 seperate PR.

File: src/test/java/microsoft/exchange/webservices/data/LazyMemberTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class LazyMemberTest {
```
Comment (serious6) [inline src/test/java/microsoft/exchange/webservices/data/LazyMemberTest.java]: please add `@RunWith(MockitoJUnit44Runner.class)`

Comment (Nowheresly) [inline src/test/java/microsoft/exchange/webservices/data/LazyMemberTest.java]: Hi @serious6 
if I add `@RunWith(MockitoJUnit44Runner.class)` to my unit test, I won't be able anymore to configure my mock in the `setUp()` method.
```java
  @Before
  public void setUp() throws Exception {

    MockitoAnnotations.initMocks(this);
    impl = new LazyMember(iLazyMember);
    // mock configuration below
    doReturn(new Object()).when(iLazyMember).createInstance();
  }
```
Any reason why you would like to add this runner ?

**Comment by Nowheresly** [comment]
PR split completed !

File: src/test/java/microsoft/exchange/webservices/data/LazyMemberTest.java (hunk 1 above)
Comment (serious6) [inline src/test/java/microsoft/exchange/webservices/data/LazyMemberTest.java]: ok.

**Comment by serious6** [comment]
@Nowheresly thanks for the contribution

**Pull request merged**